conftest.py

```diff
@@ -116,3 +116,16 @@ def _test_speedups_disable(request, settings, _test_speedups):
 
     for patcher in patchers:
         patcher.start()
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_makereport(item):
+    """
+    This function catches CommandError that are caused by merge migration errors which frequently run too long and time
+    out travis's end hook so it can't clean up it's cahce. This immediately kills all tests when merge migration errors
+    happen so that travis can end normally and clean it's cache.
+    """
+    yield
+    for exc in getattr(item, '_prepare_exc', []):
+        if 'Conflicting migrations detected; multiple leaf nodes in the migration graph:' in str(exc):
```

Comment:
Is this the only possible migration error, or more generally, are there other steps in the build setup that may prevent successful execution if they fail?

If so, a more general solution is probably warranted. It [looks like](https://docs.travis-ci.com/user/common-build-problems/#my-build-fails-unexpectedly) we may be able to toggle [`errexit`](http://tldp.org/LDP/abs/html/options.html#OPTIONSREF) in the initial steps to fail fast and hard, then unset it before running the actual test suite.